pkg/kvstore/backend.go

```diff
@@ -43,7 +43,7 @@ type backendModule interface {
 
 	// setConfigDummy must configure the backend with dummy configuration
 	// for testing purposes. This is a replacement for setConfig().
-	setConfigDummy()
+	setConfigDummy(tlsSet bool)
```

Comment:
Instead of changing this function make it always true for `consul` and false for `etcd`. Continues in [0]

Reply:
So I dont really have to change anything in etcd or other tests code, I can add just add tlsSet=true in consul ?

Reply:
@aanm Please review again.

Reply:
Yep, looks better. Now the certificates don't need to be generated for every unit test, you can add certificates already build in the repo, that will remove the requirement on having cfssl utility to run unit test.

Reply:
@aanm only issue is if server certificate is issued to a different IP then consul docker's IP., Consul will complain

Reply:
@aanm please recheck again

Reply:
>@aanm only issue is if server certificate is issued to a different IP then consul docker's IP, Consul will complain

@nirmoy I see that you have added some dummy certificates, did you manage to fix the issue? That is indeed a problem :thinking: I noticed the previous unit tests had the LDFLAGS where the consul address was a name, `consul:8500 `. Can't you create the certificate with that hostname and with `127.0.0.1`? The name `consul` will be able to run successfully in the CI, the `127.0.0.1` will be able to run Consul in our development VM.

Reply:
@aanm yes that make things easier.